string dino_base
---
string dino